### ISSUE TITLE
Friends Plugin causes wp-die on All Posts page in dashboard when a post has no title?

### ISSUE THREAD

**Issue by jeckman** [akirk/friends#208]
Running WordPress 6.2, Friends Plugin 2.5.1

If I have the friends plugin enabled, going to the all posts page in the dashboard ( /wp-admin/edit.php ) only shows 2 posts and then the page ends. 

Viewing source on the rendered html I can see there is a div of class wp-die-message that says "<p>There has been a critical error on this website. Please check your site admin email inbox for instructions.</p>"

I'm not getting any admin email inbox, and I don't see any errors in the apache log. I tried enabling debugging and looking for other info in debug bar for example but the die seems to happen before they load. 

Disabling Friends plugin makes the error go away immediately, re-enabling it makes the error return immediately 

Have not yet tried an older version, will try that text

**Comment by jeckman** [comment]
Validated same behavior with Friends 2.5.0

**Comment by jeckman** [comment]
Looks like the issue is when a post has no title - which I created by reblogging an update from a friend's activitypub stream. 

If I edit the "reblog" post to give it a title, I can re-enable the Friends plugin and not get the wp-die

**Comment by akirk** [comment]
Thanks for the report and the detective work. Do you maybe see a php filename and line number in your error log? Thank you!

**Comment by jeckman** [comment]
I didn't actually see **anything** in the error log related to these requests/pages - it was only by viewing source I was able to see the message in the wp-die-message

**Comment by jeckman** [comment]
Also happens with the autogenerated status updates ("subscribed to user's feed") that are in draft mode - since those have no title by default, trying to load the "draft" posts page causes the same error - but still no entry in the php error log that I can see

**Comment by jeckman** [comment]
That said, it doesn't seem to happen if I create manually a new post with no title in draft mode - unfortunately at this point I've deleted all my automated status posts so I can't see what was different about them

**Comment by jeckman** [comment]
Doing some debugging, finally got an error: 

Uncaught Error: Cannot use object of type WP_Error as array in /var/www/html/johneckman.com/wp-content/plugins/friends/feed-parsers/class-feed-parser-activitypub.php:484
Stack trace:
#0 /var/www/html/johneckman.com/wp-includes/class-wp-hook.php(308): Friends\Feed_Parser_ActivityPub->handle_received_activity(Array, 1, 'create')
#1 /var/www/html/johneckman.com/wp-includes/class-wp-hook.php(332): WP_Hook->apply_filters('', Array)
#2 /var/www/html/johneckman.com/wp-includes/plugin.php(517): WP_Hook->do_action(Array)
#3 /var/www/html/johneckman.com/wp-content/plugins/activitypub/includes/rest/class-inbox.php(149): do_action('activitypub_inb...', Array, 1, 'create')
#4 /var/www/html/johneckman.com/wp-includes/rest-api/class-wp-rest-server.php(1181): Activitypub\Rest\Inbox::user_inbox_post(Object(WP_REST_Request))
#5 /var/www/html/johneckman.com/wp-includes/rest-api/class-wp-rest-server.php(1028): WP_REST_Server->respond_to_request(Object(WP_REST_Request), '/activitypub/1....', Array, NULL)
#6 /var/www/html/johneckm in /var/www/html/johneckman.com/wp-content/plugins/friends/feed-parsers/class-feed-parser-activitypub.php on line 484

**Comment by jeckman** [comment]
Likely fixed by this commit https://github.com/akirk/friends/commit/2e7f01e91c7d6ad4642dbe28a8bc2281e5c585f1  

will try to find time to try that later

**Comment by jeckman** [comment]
Right, but it will let be better log what's happening there, which I wasn't getting before. 

Here's another error I caught yesterday in the log, not related to an action I took ( I took out the server path to files): 

```
[25-Apr-2023 23:09:51 UTC] PHP Fatal error:  Uncaught Error: Cannot use object of type WP_Error as array in /var/www/html/johneckman.com/wp-content/plugins/friends/feed-parsers/class-feed-parser-activitypub.php:484
Stack trace:
#0 .../wp-includes/class-wp-hook.php(308): Friends\Feed_Parser_ActivityPub->handle_received_activity(Array, 1, 'create')
#1 .../wp-includes/class-wp-hook.php(332): WP_Hook->apply_filters('', Array)
#2 .../wp-includes/plugin.php(517): WP_Hook->do_action(Array)
#3 .../wp-content/plugins/activitypub/includes/rest/class-inbox.php(149): do_action('activitypub_inb...', Array, 1, 'create')
#4 .../wp-includes/rest-api/class-wp-rest-server.php(1181): Activitypub\Rest\Inbox::user_inbox_post(Object(WP_REST_Request))
#5 .../wp-includes/rest-api/class-wp-rest-server.php(1028): WP_REST_Server->respond_to_request(Object(WP_REST_Request), '/activitypub/1....', Array, NULL)
#6 in /.../wp-content/plugins/friends/feed-parsers/class-feed-parser-activitypub.php on line 484
```

That was received activity while i was asleep ;) - hopefully getting some logs here will be helpful.

**Comment by jeckman** [comment]
Hmm. So some more interesting errors:

`[26-Apr-2023 13:34:22 UTC] WordPress database error Illegal mix of collations (utf8_general_ci,IMPLICIT) and (utf8mb4_unicode_520_ci,COERCIBLE) for operation 'like' for query 
					SELECT SQL_CALC_FOUND_ROWS  wp_posts.ID
					FROM wp_posts 
					WHERE 1=1  AND (((wp_posts.post_title LIKE '%⠺⧒👺 Himplasia Tablets - 📦 www.NetDoctor.store 📦 - Pharma without prescription 👺⧒⠺Himplasia Tablet Review, Himalaya Himplasia Tablets 30\'S%') OR (wp_posts.post_excerpt LIKE '%⠺⧒👺 Himplasia Tablets - 📦 www.NetDoctor.store 📦 - Pharma without prescription 👺⧒⠺Himplasia Tablet Review, Himalaya Himplasia Tablets 30\'S%') OR (wp_posts.post_content LIKE '%⠺⧒👺 Himplasia Tablets - 📦 www.NetDoctor.store 📦 - Pharma without prescription 👺⧒⠺Himplasia Tablet Review, Himalaya Himplasia Tablets 30\'S%')))  AND (wp_posts.post_password = '')  AND ((wp_posts.post_type = 'post' AND (wp_posts.post_status = 'publish')) OR (wp_posts.post_type = 'page' AND (wp_posts.post_status = 'publish')) OR (wp_posts.post_type = 'attachment' AND (wp_posts.post_status = 'publish')))
					
					ORDER BY (CASE WHEN wp_posts.post_title LIKE '%⠺⧒👺 Himplasia Tablets - 📦 www.NetDoctor.store 📦 - Pharma without prescription 👺⧒⠺Himplasia Tablet Review, Himalaya Himplasia Tablets 30\'S%' THEN 2 ELSE 6 END), wp_posts.post_date DESC
					LIMIT 0, 10
				 made by require('wp-blog-header.php'), wp, WP->main, WP->query_posts, WP_Query->query, WP_Query->get_posts`

Perhaps needless to say, but I have no posts with titles like that - looks like spam. 

Possible folks are trying to create posts via activity pub endpoints?

**Comment by jeckman** [comment]
Looks like what is actually happening on the view posts screen is just more out of memory errors: 

`<b>Fatal error</b>:  Allowed memory size of 2147483648 bytes exhausted (tried to allocate 20480 bytes) in <b>/var/www/html/johneckman.com/wp-includes/plugin.php</b> on line <b>203</b><br />
`

Will keep poking around

**Comment by akirk** [comment]
I am not sure this is actually in your database and Friends related but rather someone might be using the search function of your WordPress with this stuff.

**Comment by jeckman** [comment]
Right - I'm sure it is not actually making it's way into the db -

**Comment by jeckman** [comment]
Ok, so it looks like ultimately all I get is resource errors: memory limit exceeded, execution time exceeded - even if I crank those way up. 

Disabling friends plugin makes the post list load fine. 

So the question is what is happening in trying to load the posts list, with reblogs/boots in it, which causes this memory leak / loop / whatever else is happening. 

Otherwise I'm not getting any useful debug output

**Comment by ragnar76** [comment]
Got the same Problem today, this is what i got in apaches error log

```
[Thu Oct 12 21:14:25.844685 2023] [proxy_fcgi:error] [pid 1215988] [client 49.13.6.146:4178] AH01071: Got error 'PHP message: PHP Warning:  call_user_func_array() expects parameter 1 to be a valid callback, class 'Friends\\Feed_Parser_ActivityPub' does not have a method 'activitypub_activity_object_array_in_reply_to' in /home/theragnarbay/public_html/wp-includes/class-wp-hook.php on line 312'
[Thu Oct 12 21:14:36.532398 2023] [proxy_fcgi:error] [pid 1215956] [client 91.107.202.253:23343] AH01071: Got error 'PHP message: PHP Warning:  call_user_func_array() expects parameter 1 to be a valid callback, class 'Friends\\Feed_Parser_ActivityPub' does not have a method 'activitypub_activity_object_array_in_reply_to' in /home/theragnarbay/public_html/wp-includes/class-wp-hook.php on line 312'
[Thu Oct 12 21:14:42.814553 2023] [proxy_fcgi:error] [pid 1208907] [client 195.63.136.116:50746] AH01071: Got error 'PHP message: WordPress database error You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near ')\n\t\t\t\t\t\tAND relationships_author.object_id = posts.ID\n\t\t\t\t\t\tAND taxonomy_auth...' at line 10 for query SELECT relationships_post_format.term_taxonomy_id AS post_format_id, COUNT(relationships_post_format.term_taxonomy_id) AS count\n\t\t\t\t\t\tFROM wp_posts AS posts\n\t\t\t\t\t\tJOIN wp_term_relationships AS relationships_post_format\n\t\t\t\t\t\tJOIN wp_term_taxonomy AS taxonomy_author\n\t\t\t\t\t\tJOIN wp_term_relationships AS relationships_author\n\n\t\t\t\t\t\tWHERE posts.post_status IN ( 'publish', 'private' )\n\t\t\t\t\t\tAND posts.post_type IN ( 'friend_post_cache' )\n\t\t\t\t\t\tAND relationships_post_format.object_id = posts.ID\n\t\t\t\t\t\tAND relationships_post_format.term_taxonomy_id IN (  )\n\t\t\t\t\t\tAND relationships_author.object_id = posts.ID\n\t\t\t\t\t\tAND taxonomy_author.term_taxonomy_id = relationships_author.term_taxonomy_id\n\t\t\t\t\t\tAND taxonomy_author.term_id = 84\n\t\t\t\t\t\tG...', referer: https://theragnarbay.org/wp-admin/admin.php?page=add-friend
[Thu Oct 12 21:14:46.948794 2023] [proxy_fcgi:error] [pid 1187704] [client 162.55.63.189:45236] AH01071: Got error 'PHP message: PHP Warning:  call_user_func_array() expects parameter 1 to be a valid callback, class 'Friends\\Feed_Parser_ActivityPub' does not have a method 'activitypub_activity_object_array_in_reply_to' in /home/theragnarbay/public_html/wp-includes/class-wp-hook.php on line 312'
[Thu Oct 12 21:14:47.397840 2023] [proxy_fcgi:error] [pid 1215956] [client 167.235.253.94:44351] AH01071: Got error 'PHP message: PHP Warning:  call_user_func_array() expects parameter 1 to be a valid callback, class 'Friends\\Feed_Parser_ActivityPub' does not have a method 'activitypub_activity_object_array_in_reply_to' in /home/theragnarbay/public_html/wp-includes/class-wp-hook.php on line 312'
[Thu Oct 12 21:14:47.906820 2023] [proxy_fcgi:error] [pid 1215958] [client 49.13.56.118:1794] AH01071: Got error 'PHP message: PHP Warning:  call_user_func_array() expects parameter 1 to be a valid callback, class 'Friends\\Feed_Parser_ActivityPub' does not have a method 'activitypub_activity_object_array_in_reply_to' in /home/theragnarbay/public_html/wp-includes/class-wp-hook.php on line 312'
[Thu Oct 12 21:14:48.446455 2023] [proxy_fcgi:error] [pid 1216048] [client 91.107.202.253:35033] AH01071: Got error 'PHP message: PHP Warning:  call_user_func_array() expects parameter 1 to be a valid callback, class 'Friends\\Feed_Parser_ActivityPub' does not have a method 'activitypub_activity_object_array_in_reply_to' in /home/theragnarbay/public_html/wp-includes/class-wp-hook.php on line 312'
```

I'm using WordPress 6.3.1 and Friends-Plugin 2.7.9

**Comment by akirk** [comment]
@ragnar76 This is not the same, it was a typo in 2.7.9, fixed in e669e4db807bb91c482b264370b671313faf725e and I released 2.8.0. Sorry about that!

**Comment by jeckman** [comment]
I can validate this is no longer an issue on my site which was previously experiencing it, after updating to 2.8.2
